File: 2020/day_5/day_5_solution_1.pl
sub decode_row{
    my (@ops) = @_;
    my $dl = $ops[0];
        
    $dl =~ tr/B/1/;
    $dl =~ tr/F/0/;    
    return oct("0b".$dl)
}

sub decode_col{
    my (@ops) = @_;
    my $dl = $ops[0];
        
    $dl =~ tr/R/1/;
    $dl =~ tr/L/0/;    
    return oct("0b".$dl)
}

open(inputfile, "input_5.txt");

chomp(my @lines = <inputfile>);

$high_id = 0;
for $line (@lines){
    $row = substr $line, 0, 7;
    $col = substr $line, -3;
    
    $row_num = decode_row($row);
    $col_num = decode_col($col);
    $id = $row_num * 8 + $col_num;
    
    if ($id > $high_id){
        $high_id = $id;
    }
}
print "Highest ID is $high_id\n"

File: day_5/day_5_solution_1.rb
def decode(dl, sub_true, sub_false)
    dl = dl.gsub(/#{sub_true}/, "1")
    dl = dl.gsub(/#{sub_false}/, "0")
    return dl.to_i(2)
end

#test_line="BBFFBBFRLL"
#row=test_line[0..6]
#col=test_line[-3..-1]
#row_num = decode(row, "B", "F")
#col_num = decode(col, "R", "L")
#print "Binary  #{test_line} is Row #{row_num}, Col #{col_num}\n"

high_id = 0
File.open("input_5.txt").each do |line|
    line = line.strip
    row=line[0..6]
    col=line[-3..-1]
    row_num = decode(row, "B", "F")
    col_num = decode(col, "R", "L")
#    print "Binary  #{line} is Row #{row_num}, Col #{col_num}\n"
    id = row_num * 8 + col_num
    if (id > high_id)
        high_id = id
    end
end
print "Highest ID is #{high_id}\n"